bundles/mapping/mapmodule/AbstractMapModule.js

```
@@ -716,11 +736,33 @@ Oskari.clazz.define(
          */
         zoomToScale: function (scale, closest, suppressEnd) {
             var resolution = this.getResolutionForScale(scale);
+            if(!closest) {
+                // get exact resolution
+                resolution = this.getExactResolution(scale);
+                this.setResolution(resolution, suppressEnd);
+                return;
+            }
             var zoom = this.getResolutionArray().indexOf(resolution);
             if (zoom !== -1) {
                 this.setZoomLevel(zoom, suppressEnd);
             }
         },
+
+        /**
+         * Gets exact resolution
+         * @method getExactResolution
+         * @param  {Float}           scale the new scale
+         * @return {Float}           exact resolution
+         */
+        getExactResolution: function(scale) {
+            if(typeof this._getExactResolutionImpl === 'function') {
+                return this._getExactResolutionImpl(scale);
+            }
+
+            return this.getResolutionForScale(scale);
```

This should throw an exception if the exact resolution is not available.